### PR TITLE
Lower some data into the wgpu-core storages

### PR DESCRIPTION
**Pull request by i509VCB**
**Checklist**

- [X] Run `cargo clippy`.
- [X] Run `RUSTFLAGS=--cfg=web_sys_unstable_apis cargo clippy --target wasm32-unknown-unknown` if applicable.
- [ ] Add change to CHANGELOG.md. See simple instructions inside file.

**Connections**
Related to #3051 and the work to move that pull request towards using Ids instead of Boxes.

**Description**
In order for all the assoicated types in the native backend to be ids we can convert directly, we need to lower some of the data associated with some types into the tracker instead of living next to the ids.

However this is a challenge to do and this pull request is here to explore the idea or find a better alternative. There are still some types which may need data lowered into the tracker or living in a type erased form along side ids. Namely RenderPass, ComputePass and RenderBundleEncoder carry a large amount of data along.

**Testing**
Local tests appear to run fine.

### CODE REVIEW

File: wgpu-core/src/device/mod.rs [hunk 1]
```diff
@@ -3598,10 +3598,7 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         A::hub(self).buffers.label_for_resource(id)
     }
 
-    pub fn buffer_destroy<A: HalApi>(
-        &self,
-        buffer_id: id::BufferId,
-    ) -> Result<(), resource::DestroyError> {
```
Comment (ErichDonGubler) [inline wgpu-core/src/device/mod.rs:3604]: nitpick: It'd be nice to have a separate commit for moving `DestroyError` to a direct `use`. This would separate some mechanical changes from the more substantive changes in the `remove open field from CommandBuffer` commit.

Comment (i509VCB) [reply wgpu-core/src/device/mod.rs:3604]: I'll probably undo the import since every other type in that module has the qualified error type.

**Comment by i509VCB** [comment]
Going to delay this temporarily since I have a solution which is working in #3051

**Comment by cwfitzgerald** [comment]
As this MR is currently stale, I'm going to close it for triage purposes. Please feel-free to re-open there is renewed interest in landing these changes!
